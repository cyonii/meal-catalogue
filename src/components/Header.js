import Navbar from 'react-bootstrap/Navbar';
import Container from 'react-bootstrap/Container';
import { Link } from 'react-router-dom';

const Header = () => (
  <Navbar className="sticky-top" bg="light" expand="lg">
    <Container>
      <Link to="/" className="navbar-brand fw-bold">Movie Face</Link>
    </Container>
  </Navbar>
);

export default Header;
