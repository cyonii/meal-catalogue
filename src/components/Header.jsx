import { Link } from "react-router-dom";
import MovieFilter from "./MovieFilter";
import SearchForm from "./SearchForm";

const ThemeToggle = () => {
  return (
    <div className="form-check form-switch">
      <input
        className="form-check-input"
        type="checkbox"
        id="flexSwitchCheckDefault"
      />
      <label className="form-check-label" htmlFor="flexSwitchCheckDefault">
        Dark Mode
      </label>
    </div>
  );
};

const Header = () => {
  return (
    <header
      className="dark:bg-slate-900 sticky top-0 z-40 py-3"
      data-testid="header"
    >
      <div className="container w-full flex items-center justify-between">
        <Link to="/" className="text-2xl dark:text-slate-100 no-underline">
          Movie
          <b className="font-bolder">Face</b>
        </Link>

        <nav className="hidden md:flex items-center">
          <SearchForm />
          <span className="text-slate-500 mx-3">|</span>
          <MovieFilter />
        </nav>
        <button className="md:hidden px-1 py-0">
          <i className="bi bi-list text-4xl dark:text-slate-100" />
        </button>
      </div>
    </header>
  );
};

export default Header;
