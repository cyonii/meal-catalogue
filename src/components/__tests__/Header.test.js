import { screen, render } from '@testing-library/react';
import Header from '../Header';

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
}));

describe('Header', () => {
  it('should render correctly', () => {
    render(<Header />);

    expect(screen.getByText('Movie')).toBeInTheDocument();
    expect(screen.getByText('Face')).toBeInTheDocument();
    expect(screen.getByTestId('header')).toMatchSnapshot();
    expect(screen.getByTestId('filter-form')).toMatchSnapshot();
    expect(screen.getByTestId('search-form')).toMatchSnapshot();
  });
});
