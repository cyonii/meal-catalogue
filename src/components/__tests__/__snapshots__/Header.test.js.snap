// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<nav
  class="sticky-top text-white navbar navbar-expand-md navbar-dark"
  data-testid="header"
  style="background-color: rgb(31, 31, 31);"
>
  <div
    class="container"
  >
    <a
      class="navbar-brand text-white"
      href="/"
    >
      Movie
      <b
        class="fw-bold"
      >
        Face
      </b>
    </a>
    <button
      aria-controls="navbarScroll"
      aria-label="Toggle navigation"
      class="border-0 navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
    >
      <div
        class="align-items-md-center my-5 my-md-0 ms-auto navbar-nav"
      >
        <form
          class="d-flex"
          data-testid="search-form"
        >
          <input
            aria-label="Movie or show title"
            class="rounded-0 border-0 form-control"
            placeholder="Movie or show title"
            type="search"
          />
          <button
            class="rounded-0 no-outline btn btn-success"
            type="submit"
          >
            Search
          </button>
        </form>
        <span
          class="d-none d-md-block mx-md-2 text-muted"
        >
          |
        </span>
        <form
          class=""
          data-testid="filter-form"
        >
          <label
            class="form-label d-flex align-items-center mb-0 form-label"
            for="filter"
          >
            <span
              class="d-none d-lg-inline me-2"
            >
              Showing
            </span>
            <select
              aria-label="'Showing' filter form"
              class="form-control form-select rounded-0 form-select"
              id="filter"
            >
              <option
                value="popular"
              >
                Popular
              </option>
              <option
                value="topRated"
              >
                Top Rated
              </option>
              <option
                value="upcoming"
              >
                Upcoming
              </option>
              <option
                value="nowPlaying"
              >
                Now Playing
              </option>
            </select>
          </label>
        </form>
      </div>
    </div>
  </div>
</nav>
`;

exports[`Header should render correctly 2`] = `
<form
  class=""
  data-testid="filter-form"
>
  <label
    class="form-label d-flex align-items-center mb-0 form-label"
    for="filter"
  >
    <span
      class="d-none d-lg-inline me-2"
    >
      Showing
    </span>
    <select
      aria-label="'Showing' filter form"
      class="form-control form-select rounded-0 form-select"
      id="filter"
    >
      <option
        value="popular"
      >
        Popular
      </option>
      <option
        value="topRated"
      >
        Top Rated
      </option>
      <option
        value="upcoming"
      >
        Upcoming
      </option>
      <option
        value="nowPlaying"
      >
        Now Playing
      </option>
    </select>
  </label>
</form>
`;

exports[`Header should render correctly 3`] = `
<form
  class="d-flex"
  data-testid="search-form"
>
  <input
    aria-label="Movie or show title"
    class="rounded-0 border-0 form-control"
    placeholder="Movie or show title"
    type="search"
  />
  <button
    class="rounded-0 no-outline btn btn-success"
    type="submit"
  >
    Search
  </button>
</form>
`;
